File: core/src/main/java/vn/loitp/views/treeview/AlgorithmFactory.java
package vn.loitp.views.treeview;

/**
 *
 */

public class AlgorithmFactory {

    public static Algorithm createBuchheimWalker(BuchheimWalkerConfiguration configuration) {
        return new BuchheimWalkerAlgorithm(configuration);
    }

    public static Algorithm createDefaultBuchheimWalker() {
        return new BuchheimWalkerAlgorithm();
    }
}

File: core/src/main/java/com/loitpcore/views/treeView/AlgorithmFactory.kt
package com.loitpcore.views.treeView

/**
 * Created by Loitp on 04,August,2022
 * Galaxy One company,
 * Vietnam
 * +840766040293
 * freuss47@gmail.com
 */
object AlgorithmFactory {

    @Suppress("unused")
    fun createBuchheimWalker(configuration: BuchheimWalkerConfiguration?): Algorithm {
        return BuchheimWalkerAlgorithm(configuration)
    }

    @JvmStatic
    fun createDefaultBuchheimWalker(): Algorithm {
        return BuchheimWalkerAlgorithm()
    }
}
